modules/ardrivo/test/Servo.cxx

```diff
@@ -0,0 +1,83 @@
+#include <catch2/catch.hpp>
+#include <memory>
+#include <range/v3/algorithm/for_each.hpp>
+#include "BoardData.hxx"
+#include "BoardInfo.hxx"
+#include "Entrypoint.hxx"
+#include "Servo.h"
+
+void init_fake() {
+    auto loc_board_data = std::make_unique<BoardData>();
+    loc_board_data->servo_value = std::vector<std::atomic_uint8_t>(255);
+
+    auto loc_board_info = std::make_unique<BoardInfo>();
+    loc_board_info->pins_caps.resize(255);
+    ranges::for_each(loc_board_info->pins_caps, [](PinCapability& pin) { pin.pwm_able = true; });
+
+    init(loc_board_data.release(), loc_board_info.release());
+}
+
+TEST_CASE("Attach pin", "[Attach]") {
+    init_fake();
+    Servo test;
+    test.attach(1);
+    REQUIRE(test.attached());
+}
+
+TEST_CASE("Attached", "[Attached]"){
+    init_fake();
```

Comment:
No need to init_fake() every test, its better to init once and also create just one Servo.

Reply:
Yep, I agree with init_fake(). But for one Servo it will be hard to reset every time I think. Any suggestions about it?

Reply:
The servo class only stores the attached pin (which you change anyways) and the min max which you dont actually change, creating a new Servo or not really doesnt matter in this case.
All the values are written to board_data which init_fake actually resets. So at least make a global Servo object, and decide whether you really need board_data to be reset every single test.
Also maybe create a helper function to squash the write + REQUIRE(...) + board_data check.

100 comments :100:

Reply:
No but the attched pin need to be reset. Cause it is -1 if it is not attached. Maybe make a helper function to reset if we really want to do it.

Reply:
Objection: one `Servo` per test-case is intended, since `Servo` is the class under test, however test cases should be full scenarios, where single functions are tested in their own sections